core/multi_map.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import "time"
+
+// MultiMap is a specialized map whose keys can be associated with multiple values.
+type MultiMap interface {
+	// IDistributedObject is the base interface for all distributed objects.
+	IDistributedObject
+
+	// Stores a key-value pair in the multi-map.
+	// It returns true if size of the multi-map is increased, false if the multi-map
+	// already contains the key-value pair.
+	Put(key interface{}, value interface{}) (increased bool, err error)
```

Comment:
I think it would be better to use a more descriptive return name here, maybe "*changed*" ?

Reply:
I think "*increased*" is descriptive enough.

Reply:
`changed` is not correct as it includes decrease.

We can use something like `mapSizeIncreased`